static/app/components/compactSelect/utils.tsx

```diff
@@ -40,7 +40,10 @@ export function getItemsWithKeys<Value extends SelectKey>(
       };
     }
 
-    return {...item, key: getEscapedKey(item.value)};
+    const existingKey =
```

Comment:
Checking for an existing key here to prevent the key from being escaped more than once if this function were called multiple times. Not sure if that is actually done, but wanted to be careful